src/pyuvsim/telescope.py

```diff
@@ -61,112 +60,69 @@ class BeamList:
-        Passing in a mixture of strings and objects will error.
-    uvb_read_kwargs : dict (optional)
-        A nested dictionary that can contain parameters to pass to the UVBeam
-        read method for each UVBeam in the beam dict. The top-level keys are
-        beam_id integers, the value for each beam id is a dict with the kwargs
-        specified for each beam. This can include ``file_type`` or any other
-        parameter accepted by the UVBeam.read method. Note that this can in
-        principal overlap with entries in select_params. Entries in uvb_read_kwargs
-        will supercede anything that is also in select_params.
-    select_params : dict (optional)
-        A dictionary that can contain parameters for selecting parts of the beam
-        to read. Example keys include ``freq_range`` and ``za_range``. Note that
-        these will only be used for UVBeam readable files and they apply to all
-        such beams unless they are superceded by uvb_read_kwargs.
-    check : bool
-        Whether to perform a consistency check on the beams (i.e. asserting that several
-        of their defining parameters are the same for all beams in the list).
-    force_check : bool
-        The consistency check is only possible for object-beams (not string mode). If
-        `force_check` is True, it will convert beams to object-mode to force the check
-        to run (then convert back to string mode).
-
-    Raises
-    ------
-    ValueError
-        For an invalid beam_list (mix of strings and objects).
-
-    Notes
-    -----
-    If an object beam is added while in string mode, it will be converted to a string
-    before being inserted, and vice versa. ***In order to be converted to a string, a UVBeam
-    needs to have the entry 'beam_path' in its extra_keywords, giving the path to the beam
-    file it came from.***
+    beams : list
+        A list of pyuvdata UVBeam or AnalyticBeam objects.
+    beam_type : str
+        Desired beam type, either "efield" or "power", defaults to "efield".
+    spline_interp_opts : dict
+        Provide options to numpy.RectBivariateSpline. This includes spline
+        order parameters `kx` and `ky`, and smoothing parameter `s`.
+        Only applies UVBeams and for `az_za_simple` interpolation.
+    freq_interp_kind : str
+        Interpolation method to use for frequencies. See scipy.interpolate.interp1d
+        for details. Defaults to "cubic".
+    peak_normalize : bool
+        Option to peak normalize UVBeams in the beam list. This is required to
+        be True for the pyuvsim simulator. Defaults to True.
 
     """
 
-    _float_params = {
-        "sig": "sigma",
-        "diam": "diameter",
-        "reff": "ref_freq",
-        "ind": "spectral_index",
-    }
+    beams: InitVar([UVBeam | AnalyticBeam])
+    _: KW_ONLY
+    beam_type: Literal["efield", "power"] | None = "efield"
+    spline_interp_opts: dict[str:int] | None = None
```

Comment:
I thought the specification of a dict type was in the format `dict[keytype, valtype]` (i.e. comma instead of colon)?